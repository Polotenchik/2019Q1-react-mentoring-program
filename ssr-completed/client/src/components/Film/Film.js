import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import styled from 'styled-components';

const Film = ({ info }) => (
  <FilmLink to={`/film/${info.id}`} key={info.id}>
    <Img src={info.poster_path} />
    <InfoWrapper>
      <Title>{info.title}</Title>
      <span>{info.genres.join(', ')}</span>
      <FilmYearRelease>
        {info.release_date.split('-')[0]}
      </FilmYearRelease>
    </InfoWrapper>
  </FilmLink>
);

export default Film;

Film.propTypes = {
  info: PropTypes.shape({
    id: PropTypes.number,
    title: PropTypes.string,
    tagline: PropTypes.string,
    image: PropTypes.string,
    vote_average: PropTypes.number,
    vote_count: PropTypes.number,
    releasedate: PropTypes.number,
    poster_path: PropTypes.string,
    overview: PropTypes.string,
    budget: PropTypes.number,
    revenue: PropTypes.number,
    genres: PropTypes.array,
    runtime: PropTypes.number,
  }),
};

const FilmLink = styled(Link).attrs({
  className: 'film-link',
})`
    position: relative;
    width: 30%;
    height: 0;
    padding-top: 55%;
    text-decoration: none;
    box-shadow: 0 4px 11px 0 grey;
    margin-bottom: 30px;
    cursor: pointer;  
      
  @media (max-width: 900px) {
    width: 48%;
    padding-top: 88%;
  }
  
  @media (max-width: 500px) {
    width: 100%;
    padding-top: 183%;
  }    
`;

const Img = styled.img`
    position: absolute;
    top: 0;
    left: 0;
    width: 100%;
    height: 80%;
    padding: 16px;
    box-sizing: border-box;
`;

const InfoWrapper = styled.div`
    display: flex;
    flex-direction: column;
    justify-content: space-between;
    position: absolute;
    bottom: 0;
    width: 100%;
    height: 20%;
    padding: 0 16px 16px ;
    box-sizing: border-box;
`;

const Title = styled.h3`
    text-transform: uppercase;
    padding: 0 40px 0 0;
    margin: 0;
    overflow: hidden;
    text-overflow: ellipsis;
    display: -webkit-box;
    -webkit-line-clamp: 2;
    -webkit-box-orient: vertical;
`;

const FilmYearRelease = styled.span`
    position: absolute;
    top: 0;
    right: 16px;
    border: 1px solid black;
    padding: 4px;
    font-size: 12px;
`;
