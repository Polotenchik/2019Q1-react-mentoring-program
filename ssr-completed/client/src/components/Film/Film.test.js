import React from 'react';
import { render } from 'enzyme';
import Film from './Film';

describe('Film Snapshot', () => {
    it('Film Component should render correctly with given props', () => {
        const info={
            "id":5924,
            "title":"Papillon",
            "tagline":"The greatest adventure of escape!",
            "vote_average":7.8,
            "vote_count":564,
            "release_date":"1973-12-13",
            "poster_path":"https://image.tmdb.org/t/p/w500/xY7k2M9eElJnLSYwqY85Xon5z6m.jpg",
            "overview":"A man befriends a fellow criminal as the two of them begin serving their sentence on a dreadful prison island, which inspires the man to plot his escape.",
            "budget":12000000,
            "revenue":53267000,
            "genres":["Crime","Drama"],
            "runtime":151
        };
        const albumClickHandler=jest.fn();
        const component = render(<Film info={info} albumClickHandler={albumClickHandler} />);
        expect(component).toMatchSnapshot();
    });
});

