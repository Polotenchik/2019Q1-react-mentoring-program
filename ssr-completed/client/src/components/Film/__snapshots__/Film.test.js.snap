// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Film Snapshot Film Component should render correctly with given props 1`] = `
<a
  class="sc-bdVaJa film-link dQMKLa"
>
  <img
    class="sc-bwzfXH ifbPgJ"
    src="https://image.tmdb.org/t/p/w500/xY7k2M9eElJnLSYwqY85Xon5z6m.jpg"
  />
  <div
    class="sc-htpNat eUMQJL"
  >
    <h3
      class="sc-bxivhb gEscEi"
    >
      Papillon
    </h3>
    <span>
      Crime, Drama
    </span>
    <span
      class="sc-ifAKCX cCwMba"
    >
      1973
    </span>
  </div>
</a>
`;
